tests/posts.test.js

```diff
@@ -0,0 +1,181 @@
+const request = require("supertest");
+const initApp = require("../server.js");
+const mongoose = require("mongoose");
+const postModel = require("../models/posts_model.js");
+
+let app;
+beforeAll(async () => {
+  app = await initApp();
+});
```

Comment:
should return 500 for a null sender in getbysender (meaning a sender not sent in the api)
should return 500 when there is a server error in get all posts